### ISSUE TITLE
No module named migrate.versioning.api at startup

### ISSUE THREAD

**Issue by haansn08** [pytrainer/pytrainer#37]
Just installed pytrainer with `sudo python setup.py install`, but the program didn't start up.

Traceback (most recent call last):
  File "/bin/pytrainer", line 96, in <module>
    from pytrainer.main import pyTrainer
  File "/usr/lib/python2.7/site-packages/pytrainer/main.py", line 36, in <module>
    from upgrade.data import initialize_data
  File "/usr/lib/python2.7/site-packages/pytrainer/upgrade/data.py", line 23, in <modu$
    from pytrainer.upgrade.migratedb import MigratableDb
  File "/usr/lib/python2.7/site-packages/pytrainer/upgrade/migratedb.py", line 19, in $
    from migrate.versioning.api import db_version, upgrade, version, version_control
ImportError: No module named migrate.versioning.api`

**Comment by Charlie37** [comment]
How do you install python-migrate ? I am on Fedora 21 and the package is not found when typing pip3 install python-migrate.

**Comment by azapletal** [comment]
@Charlie37 Just install the distribution package.

**Comment by Charlie37** [comment]
You mean with Fedora ? What is the command please ?

**Comment by azapletal** [comment]
`dnf install python-migrate`
maybe if you're still on F21 try
`yum install python-migrate`

but these are basics ;)

**Comment by Charlie37** [comment]
OK thanks :) Yes, we're still on F21.

The dnf command fails:
`Error: nothing provides python-sqlparse needed by python-migrate-0.9.4-1.fc21.noarch`

The yum command ends with:
```
Error: Package: python-migrate-0.9.4-1.fc21.noarch (updates)
           Requires: python-tempita

Error: Package: python-migrate-0.9.4-1.fc21.noarch (updates)
           Requires: python-sqlparse
```

With the --skip-broken option added, the command ends with: 

```
Packages skipped because of dependency problems:
    python-migrate-0.9.4-1.fc21.noarch from updates
    python-sqlalchemy-0.9.10-1.fc21.x86_64 from updates
```

The error is now that 'migrate is not a package', although it's present in python3.4/site-packages. 
We've commented here: https://github.com/miguelgrinberg/microblog/issues/18

Thanks again for your insights

**Comment by azapletal** [comment]
I see. Well, support of F21 ended last december - therefore maybe the mess, dunno ;)
On F23 this installed flawless as far as I remember. You could think about an upgrade (for security reasons always recommended).

**Comment by Charlie37** [comment]
@azapletal OK, thanks a lot. Not sure we could upgrade at the moment, but we'll check this out.

**Comment by eternal1025** [comment]
ubantu
sudo pip install -i http://mirrors.aliyuncs.com/pypi/simple SQLAlchemy-migrate

Windows
pip install  SQLAlchemy-migrate

**Comment by viiru-** [comment]
This issue was somewhat accidentally fixed in commit 4ff870b359d3136c7ea11d7ff69a36bca36b73bb.
